src/Microsoft.TemplateEngine.Cli/ExtendedTemplateEngineHost.cs

```diff
@@ -144,5 +146,26 @@ private bool GlobalJsonFileExistsInPath
                 return found;
             }
         }
+
+        public void RebuildCacheFromSettingsIfNotCurrent(IEngineEnvironmentSettings environmentSettings)
+        {
+            TemplateCache cache = new TemplateCache(environmentSettings);
+            cache.LoadTemplateCacheForLocale(Locale, out string cacheVersion);
+
+            if (TemplateCache.CheckIfCacheVersionIsCurrent(cacheVersion))
+            {
+                return;
+            }
+
+            cache.CacheVersion = TemplateCache.CurrentCacheVersion;
+            cache.TemplateInfo.Clear();
+
+            foreach (MountPointInfo mountPoint in environmentSettings.SettingsLoader.MountPoints)
```

Comment:
Should we read the information from the templates currently in the cache (if we can) so that we don't end up scanning mount points that contain only components for templates?